Application/EdFi.Ods.Tests/EdFi.Ods.Api/Controllers/TokenControllerTests.cs

```diff
@@ -1,1626 +1,1652 @@
-    //         {
-    //             private IClientAppRepo _clientAppRepo;
-    //             private IApiClientAuthenticator _apiClientAuthenticator;
-    //             private TokenController _controller;
-    //
-    //             private HttpResponseMessage _actualResponseMessage;
-    //             private JObject _actualJsonContent;
-    //
-    //             protected override void Arrange()
-    //             {
-    //                 _clientAppRepo = mocks.StrictMock<IClientAppRepo>();
-    //                 _apiClientAuthenticator = _apiClientAuthenticatorHelper.Mock(mocks);
-    //
-    //                 _controller = CreateTokenController(_clientAppRepo, _apiClientAuthenticator);
-    //             }
-    //
-    //             protected override void Act()
-    //             {
-    //                 _actualResponseMessage = _controller.Post(
-    //                         new TokenRequest
-    //                         {
-    //                             Client_id = "clientId", Grant_type = "client_credentials"
-    //                         })
-    //                     .ExecuteAsync(new CancellationToken())
-    //                     .Result;
-    //
-    //                 string actualContent = _actualResponseMessage.Content.ReadAsStringAsync()
-    //                     .Result;
-    //
-    //                 _actualJsonContent = JObject.Parse(actualContent);
-    //             }
-    //
-    //             [Assert]
-    //             public void Should_return_HTTP_status_of_BadRequest()
-    //             {
-    //                 Assert.That(_actualResponseMessage.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
-    //             }
-    //
-    //             [Assert]
-    //             public void Should_return_a_single_valued_response_with_an_error_indicating_invalid_client()
-    //             {
-    //                 Assert.That(
-    //                     _actualJsonContent.Properties()
-    //                         .Count(),
-    //                     Is.EqualTo(1),
-    //                     _actualJsonContent.ToString());
-    //
-    //                 Assert.That(
-    //                     _actualJsonContent["error"]
-    //                         .Value<string>(),
-    //                     Is.EqualTo("invalid_client"));
-    //             }
-    //
-    //             [Assert]
-    //             public void Should_not_include_any_cache_headers()
-    //             {
-    //                 Assert.That(_actualResponseMessage.Headers.CacheControl, Is.Null);
-    //                 Assert.That(_actualResponseMessage.Headers.Pragma, Is.Empty);
-    //             }
-    //
-    //             public override void RunOnceAfterAll()
-    //             {
-    //                 _clientAppRepo.VerifyAllExpectations();
-    //             }
-    //         }
-    //     }
-    // }
+    public class When_calling_the_token_controller
```

Comment:
we should wrap the all the tests in a class named TokenControllerTests. It should have the [TestFixture] attribute so the tests are aggregated. 

Note this is a nice to have